### PR TITLE
Layoutlm onnx support

### PR DESCRIPTION
**Pull request by nishprabhu**
# What does this PR do?

This PR extends ONNX support to LayoutLM as explained in https://huggingface.co/transformers/serialization.html?highlight=onnx#converting-an-onnx-model-using-the-transformers-onnx-package

Fixes Issue  #13300

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [X] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [X] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@mfuntowicz @NielsRogge

### CODE REVIEW

File: src/transformers/models/layoutlm/configuration_layoutlm.py [hunk 1]
```diff
@@ -125,3 +131,84 @@ def __init__(
...
+class LayoutLMOnnxConfig(OnnxConfig):
+
+    DEFAULT_MAX_2D_POSITION = 1000
+
+    @property
+    def inputs(self) -> Mapping[str, Mapping[int, str]]:
+        return OrderedDict(
+            [
+                ("input_ids", {0: "batch", 1: "sequence"}),
+                ("bbox", {0: "batch", 1: "sequence"}),
```
[the remark this answers is not in the document]
Comment (NielsRogge) [reply src/transformers/models/layoutlm/configuration_layoutlm.py:145]: Oh how did you know that? Is this mentioned in the docs?

[the remark this answers is not in the document]
Comment (nishprabhu) [reply src/transformers/models/layoutlm/configuration_layoutlm.py:145]: Its not explicitly mentioned in the docs. But the tutorial link you sent shows the config for Bert, and that does not specify the static dimension in the output (the 768 dimension in "last_hidden_state" and "pooler_output"). So I went through the code in src/transformers/onnx to verify this. The dimensions returned by the config file are passed on to the export function provided by PyTorch in the torch.onnx module as shown in src/transformers/onxx/conver.py:

```
    export(
        model,
        (model_inputs,),
        f=output.as_posix(),
        input_names=list(config.inputs.keys()),
        output_names=onnx_outputs,
        dynamic_axes={name: axes for name, axes in chain(config.inputs.items(), config.outputs.items())},
        do_constant_folding=True,
        use_external_data_format=config.use_external_data_format(model.num_parameters()),
        enable_onnx_checker=True,
        opset_version=opset,
    )

```
As you can see, the shapes provided by the config file are passed to the dynamic_axes parameter in the function call.

I also verified the change by converting a LayoutLM model to ONNX- the verification step that runs during conversion confirmed that the outputs of the converted model are similar to the original. Visualizing the converted ONNX model using Netron also shows that the input and output sizes are as follows:

```
Inputs:     
------     
name: input_ids    
type: int64[batch,sequence]    
    
name: bbox    
type: int64[batch,sequence,4]    
    
name: attention_mask    
type: int64[batch,sequence]    
    
name: token_type_ids    
type: int64[batch,sequence]    
    
Outputs:     
-------     
name: last_hidden_state    
type: float32[batch,sequence,768]    
    
name: pooler_output    
type: float32[batch,768]
```

I do think that this should be explicitly mentioned in the docs though, so that it will be easier for anyone trying to convert custom models.

[the remark this answers is not in the document]
Comment (mfuntowicz) [reply src/transformers/models/layoutlm/configuration_layoutlm.py:145]: @nishprabhu you're totally right, this is about mapping the dynamic axis for each tensor (_i.e. axis we don't know beforehand or can change between two successive calls_).

In this case, if the bounding box dimensions are constant (x1, y2, x2, y2) then ONNX will infer and reports this in the traced model 👍🏻

[the remark this answers is not in the document]
Comment (mfuntowicz) [reply src/transformers/models/layoutlm/configuration_layoutlm.py:145]: We will update the documentation to better reflect this 👍🏻

**Comment by mfuntowicz** [comment]
@nishprabhu would you mind closing this PR and opening a new one, without any additional changes?

Github has a weird issue making the diff almost impossible to review.

Sorry for the inconvenience, please let us know if you need any help.

File: src/transformers/models/layoutlm/configuration_layoutlm.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (NielsRogge) [reply src/transformers/models/layoutlm/configuration_layoutlm.py:143]: That's actually interesting to know, I've never worked with ONNX, but this seems indeed pretty important.

**Comment by nishprabhu** [comment]
Sure, @mfuntowicz 
I'll open a new PR with the changes.
